File: Examples/CSharp/SimpleElastix.cs
using System;
using itk.simple;

namespace itk.simple.examples {
    class SimpleElastix {
        static void Main(string[] args) {
            try {
                if (args.Length < 4) {
                    Console.WriteLine("Usage: SimpleElastix <fixedImage> <movingImage> <parameterFile> <output>");
                    return;
                }
                // Instantiate SimpleElastix
                ElastixImageFilter silx;

                // Read input 
                ImageFileReader reader = new ImageFileReader();
                reader.SetFileName(args[0]);
                silx.SetFixedImage(reader.Execute());
                reader.SetFileName(args[1]);
                silx.SetMovingImage(reader.Execute());
                silx.SetParameterMap(ReadParameterFile(args[2]));

                // Perform Registration
                silx.LogToConsoleOn();
                silx.Execute();

                // Write output image
                ImageFileWriter writer = new ImageFileWriter();
                writer.SetFileName(args[3]);
                writer.Execute(silx.GetResultImage());

            } catch (Exception e) {
                Console.WriteLine(e);
            }
        }
    }
}

File: Examples/Java/SimpleElastix.java
import org.itk.simple.*;

class SimpleElastix {
 
  public static void main(String[] argv) {

    // If you get "no SimpleITKJava in java.library.path" point java.library.path to lib/libSimpeITKJava.jnilib

    if ( argv.length < 4 ) {
      System.out.println("Usage: java -cp SimpleElastix.jar SimpleElastix <fixedImage> <movingImage> <parameterFile> <outputFilename>");
      return;
    }

    // Instantiate SimpleElastix
    ElastixImageFilter elastixImageFilter = new ElastixImageFilter();

    // Read input
    ImageFileReader reader = new ImageFileReader();
    reader.setFileName(argv[0]);
    elastixImageFilter.setFixedImage(reader.execute());
    reader.setFileName(argv[1]);
    elastixImageFilter.setMovingImage(reader.execute());
    elastixImageFilter.setParameterMap(elastixImageFilter.readParameterFile(argv[2]));

    // Perform registration
    elastixImageFilter.logToConsoleOn();
    elastixImageFilter.execute();

    // Write result image
    ImageFileWriter writer = new ImageFileWriter();
    writer.setFileName(argv[3]);
    writer.execute(elastixImageFilter.getResultImage()); 

    elastixImageFilter.printParameterMap(elastixImageFilter.getTransformParameterMap());

  }

}
